test/functional/update/verify-fullfile-hash/test.bats

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bats
+
+load "../../swupdlib"
+
+f1=520f83440d3dddc25ad09ca858b9c669245f82d3181a45cdfe793aac9dd1fb15
+
+setup() {
+  clean_test_dir
+  tar -C "$DIR/web-dir/10" -cf "$DIR/web-dir/10/Manifest.MoM.tar" Manifest.MoM Manifest.MoM.signed
+  tar -C "$DIR/web-dir/10" -cf "$DIR/web-dir/10/Manifest.os-core.tar" Manifest.os-core Manifest.os-core.signed
+  tar -C "$DIR/web-dir/10" -cf "$DIR/web-dir/10/Manifest.test-bundle.tar" Manifest.test-bundle Manifest.test-bundle.signed
+  tar -C "$DIR/web-dir/100" -cf "$DIR/web-dir/100/Manifest.MoM.tar" Manifest.MoM Manifest.MoM.signed
+  tar -C "$DIR/web-dir/100" -cf "$DIR/web-dir/100/Manifest.test-bundle.tar" Manifest.test-bundle Manifest.test-bundle.signed
+  sudo chown root:root "$DIR/web-dir/100/files/$f1"
+  tar -C "$DIR/web-dir/100/files" -cf "$DIR/web-dir/100/files/$f1.tar" $f1
+}
+
+teardown() {
+  pushd "$DIR/web-dir/10"
+  rm *.tar
+  popd
+  pushd "$DIR/web-dir/100"
+  rm *.tar
+  popd
+  pushd "$DIR/web-dir/100/files"
+  rm *.tar
+  popd
+  sudo chown $(ls -l "$DIR/test.bats" | awk '{ print $3 ":" $4 }') "$DIR/web-dir/100/files/$f1"
+}
+
+@test "update fullfile hashes verified" {
+  run sudo sh -c "$SWUPD update $SWUPD_OPTS"
+
+  echo "$output"
+  [ "${lines[2]}" = "Attempting to download version string to memory" ]
+  [ "${lines[3]}" = "Update started." ]
+  [ "${lines[4]}" = "Querying server version." ]
+  [ "${lines[5]}" = "Attempting to download version string to memory" ]
+  [ "${lines[6]}" = "Preparing to update from 10 to 100" ]
+  [ "${lines[7]}" = "Querying current manifest." ]
+  [ "${lines[8]}" = "Querying server manifest." ]
+  [ "${lines[9]}" = "Downloading test-bundle pack for version 100" ]
+  [ "${lines[10]}" = "Statistics for going from version 10 to version 100:" ]
+  [ "${lines[11]}" = "    changed manifests : 1" ]
+  [ "${lines[12]}" = "    new manifests     : 0" ]
+  [ "${lines[13]}" = "    deleted manifests : 0" ]
+  [ "${lines[14]}" = "    changed files     : 1" ]
+  [ "${lines[15]}" = "    new files         : 0" ]
+  [ "${lines[16]}" = "    deleted files     : 0" ]
+  [ "${lines[17]}" = "Starting download of remaining update content. This may take a while..." ]
+  [ "${lines[18]}" = "Finishing download of update content..." ]
+  [ ! -f "$DIR/target-dir/foo" ]
```

Comment:
We'd have to help the string because it contains the full path with test
directory. I wanted to avoid that bit.
On May 19, 2016 5:26 PM, "Patrick McCarty" <notifications@github.com> wrote:

> In test/functional/update/verify-fullfile-hash/test.bats
> <https://github.com/clearlinux/swupd-client/pull/72#discussion_r63975321>:
>
> > +  [ "${lines[4]}" = "Querying server version." ]
> > +  [ "${lines[5]}" = "Attempting to download version string to memory" ]
> > +  [ "${lines[6]}" = "Preparing to update from 10 to 100" ]
> > +  [ "${lines[7]}" = "Querying current manifest." ]
> > +  [ "${lines[8]}" = "Querying server manifest." ]
> > +  [ "${lines[9]}" = "Downloading test-bundle pack for version 100" ]
> > +  [ "${lines[10]}" = "Statistics for going from version 10 to version 100:" ]
> > +  [ "${lines[11]}" = "    changed manifests : 1" ]
> > +  [ "${lines[12]}" = "    new manifests     : 0" ]
> > +  [ "${lines[13]}" = "    deleted manifests : 0" ]
> > +  [ "${lines[14]}" = "    changed files     : 1" ]
> > +  [ "${lines[15]}" = "    new files         : 0" ]
> > +  [ "${lines[16]}" = "    deleted files     : 0" ]
> > +  [ "${lines[17]}" = "Starting download of remaining update content. This may take a while..." ]
> > +  [ "${lines[18]}" = "Finishing download of update content..." ]
> > +  [ ! -f "$DIR/target-dir/foo" ]
>
> Initially, I was confused about what this test's purpose was, but then I
> saw the full $output contents and understand now...
>
> Should we test that the "Error: File content hash mismatch ..." message is
> in fact printed, since it proves we detect this error condition?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/clearlinux/swupd-client/pull/72/files/e0b859448a3e41e0e6fb78acd26768b184e81651..9f4feee501afad0a43f3fb42dca61107bab7f131#r63975321>
>